src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/util.py

```diff
@@ -105,13 +109,17 @@ def create_short_lived_container_sas(account_name, account_key, container):
 
 def create_short_lived_share_sas(account_name, account_key, share):
     from datetime import datetime, timedelta
-    SharedAccessSignature, BlobPermissions = get_sdk(ResourceType.DATA_STORAGE,
-                                                     'sharedaccesssignature#SharedAccessSignature',
-                                                     'blob.models#BlobPermissions')
+    if supported_api_version(ResourceType.DATA_STORAGE, min_api='2017-04-17'):
+        SharedAccessSignature = get_sdk(ResourceType.DATA_STORAGE, 'FileSharedAccessSignature',
+                                        mod='file.sharedaccesssignature')
+    else:
+        SharedAccessSignature = get_sdk(ResourceType.DATA_STORAGE, 'SharedAccessSignature',
+                                        mod='sharedaccesssignature')
+    FilePermission = get_sdk(ResourceType.DATA_STORAGE, 'file.models#FilesPermissions')
 
     expiry = (datetime.utcnow() + timedelta(days=1)).strftime('%Y-%m-%dT%H:%M:%SZ')
     sas = SharedAccessSignature(account_name, account_key)
-    return sas.generate_share(share, permission=BlobPermissions(read=True), expiry=expiry, protocol='https')
+    return sas.generate_share(share, permission=FilePermission(read=True), expiry=expiry, protocol='https')
```

Comment:
Why did the SDK change require FilePermission instead of BlobPermissions now?

Reply:
It was obviously a bug on our side before.